modules/wayfire/hm.nix

```
@@ -0,0 +1,64 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  options.stylix.targets.wayfire.enable =
+    config.lib.stylix.mkEnableTarget "wayfire" true;
+
+  config =
+    lib.mkIf (config.stylix.enable && config.stylix.targets.wayfire.enable)
+      (
+        let
+          inherit (config.lib.stylix) colors;
+          rgb = rgb: ''\#${rgb}ff'';
+
+          wayfireCfg = config.wayland.windowManager.wayfire;
+
+          backgroundImage = toString (
+            pkgs.runCommand "wayfire-background.png" { } ''
+              ${lib.getExe pkgs.imagemagick} ${config.stylix.image} $out
+            ''
+          );
```

AFAIK, `toString` does not prevent the GC from removing the path, causing it to be invalid afterwards. However, I am unsure whether `pkgs.runCommand` prevents that in this scope. Either way, explicitly preventing GC by wrapping it in `"${}"` should be safer:

```suggestion
          backgroundImage = "${
            pkgs.runCommand "wayfire-background.png" { } ''
              ${lib.getExe pkgs.imagemagick} ${config.stylix.image} $out
            ''
          }";
```

Also, the following question has not been addressed yet:

> > Wayfire currently isnt smart enough to understand what file type an image has if it doesnt have a file ending and we cannot depend on that being set.
>
> This adds the `.png` extension to the `realpath` of the `/nix/store/<HASH>-wayfire-background.png` symlink. Does the `convert "$stylix_image" result` command actually do something? Also, does this work when `stylix.image` is not a PNG?
>
> -- https://github.com/danth/stylix/pull/765#discussion_r1921092389